src/Sarif/FileRegionsCache.cs

```diff
@@ -156,15 +156,42 @@ internal Region ConstructMultilineContextSnippet(Region inputRegion, Uri uri)
                 return null;
             }
 
+            const int maxSnippetLength = 512;
+            const int reasonableSnippetLength = 256;
+            const int smallSnippetLength = 128;
+
+            // If this is big enough, we don't need to create a new snippet.
+            if (inputRegion.Snippet?.Text.Length >= reasonableSnippetLength)
+            {
+                return null;
+            }
```

Comment:
shouldn't we actually be checking the region size, not the snippet size? basically, if a region exceeds a certain size, no context region is required. so for this case, the region should have the snippet. your code will permit the following case:

region is 10k in size but has no snippet
we then create a context region + snippet which is smaller than the region! this breaks the SARIF spec. :)

Reply:
changed to the charLength.

---
In reply to: [578048420](https://github.com/microsoft/sarif-sdk/pull/2288#discussion_r578048420) [](ancestors = 578048420)